libs/s25main/ai/random.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2005 - 2025 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "helpers/random.h"
+#include <random>
+
+namespace AI {
+
+std::minstd_rand& getRandomGenerator();
+
+template<typename T>
+T randomValue(T min = std::numeric_limits<T>::min(), T max = std::numeric_limits<T>::max())
+{
+    return helpers::randomValue(getRandomGenerator(), min, max);
+}
+
+// Return a random bool:
+//   random()           ... will return true|false with 50% chance each
+//   random(15)         ... will return true in 1/15 of the cases
+inline bool random(unsigned total = 1u)
+{
+    return helpers::randomValue(getRandomGenerator(), 0u, total) == 0u;
```

Comment:
If the intention is "1 out of `total` cases" you need 
```suggestion
    return randomValue(1u, total) == 1u;
```
(with less code duplication using the function above)

Remember that the parameters are `minValue, maxValue`, inclusive! Otherwise generation a `UINT_MAX` would be impossible.

And I'd suggest to use the below overload instead if that already means `chance out of total` as you simply have `chance == 1`. Hence the suggestion to use a default parameter instead of an overload.

Or is there an advantage in having 2?

Reply:
No, with the `<= chance` check there is no need for two functions anymore.